spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -77,23 +78,36 @@ public DatastoreTemplate(Datastore datastore,
 	}
 
 	@Override
-	public <T> void save(T instance) {
-		this.datastore.put(convertToEntity(instance));
+	public <T> T save(T instance) {
+		Entity saved = this.datastore.put(convertToEntity(instance));
+		DatastorePersistentEntity datastorePersistentEntity = this.datastoreMappingContext
+				.getPersistentEntity(instance.getClass());
+		datastorePersistentEntity.getPropertyAccessor(instance).setProperty(
```

Comment:
I don't understand the need for all this logic. It looks like `getKeyFromObject` already updates the ID property of the `instance`. You should be able to just return `instance` after the operation and it will have the id allocated. What am I missing?

Reply:
You're right. I forgot that it did that!. Will fix